### PR TITLE
V1beta2 report resource

### PR DESCRIPTION
**Pull request by snehagunta**
## Describe your changes

- ...

## Ticket reference (if applicable)
Fixes #

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

## Summary by Sourcery

Add v1beta2 reporting resource support by defining new data models, repository implementations, database migrations, transaction utilities, and comprehensive unit tests.

New Features:
- Introduce v1beta2 models: Resource, RepresentationReference (with unique composite index and Postgres covering indexes), CommonRepresentation, ReporterRepresentation, ResourceWithReferences aggregate, and ReporterRepresentationId.
- Implement real and in-memory fake repositories for ResourceWithReferences (including find-by-reporter lookup), CommonRepresentation, and ReporterRepresentation with transactional create operations.
- Integrate v1beta2 models into the GORM auto-migration setup.

Enhancements:
- Add WithTx helper to manage transactions uniformly across v1beta2 repositories.

Tests:
- Add extensive unit tests for real and fake v1beta2 repositories covering creation, duplicate key handling, search functionality, persistence, transaction commits, reset, and getAll behaviors.

### CODE REVIEW

**Comment by snehagunta** [comment]
@sourcery-ai review

File: internal/biz/model/v1beta2/representation_reference.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			ON representation_references (resource_id)
+			INCLUDE (
+				reporter_type,
+				representation_version,
+				generation,
+				tombstone);
+		`).Error
+		if err != nil {
+			return fmt.Errorf("failed to create rep_ref_resource_id_covering_idx: %w", err)
+		}
```
Comment (tylercreller) [inline internal/biz/model/v1beta2/representation_reference.go:60]: We should avoid doing manual migrations like this while having gorm automigrations enabled. I believe every time we roll out this code, gorm automigrate will remove the indexes and then this code will re-add them, and we don't want that running on every deploy.

https://gorm.io/docs/migration.html

There are strategies we can use for manual migrations, but we would need to consider disabled automigrate.
